application_template/Mentor_application.md

```diff
@@ -0,0 +1,48 @@
+# Our Mentee Application Template
+
+Hello! Thank you for your interest in being a part of Galaxy Mentoring Network!
+
+**Kindly download this template so that you can have a draft before you tranfer your responses to the form.**
```

Comment:
Download will likely not working there. Maybe better to say "Copy / paste"

Reply:
Thinking out loud, should we embed the templates (doc) to the application guideline page?